geopandas/tests/test_extension_array.py

```diff
@@ -334,7 +335,74 @@ class TestConstructors(extension_tests.BaseConstructorsTests):
 
 
 class TestReshaping(extension_tests.BaseReshapingTests):
-    pass
+
+    # NOTE: this test is copied from pandas/tests/extension/base/reshaping.py
+    # because starting with pandas 3.0 the assert_frame_equal is strict regarding
+    # the exact missing value (None vs NaN)
+    # Our `result` uses None, but the way the `expected` is created results in
+    # NaNs (and specifying to use None as fill value in unstack also does not
+    # help)
+    # -> the only change compared to the upstream test is marked
+    @pytest.mark.parametrize(
```

Comment:
This is unfortunate that we have to copy the whole test, but I don't directly see any other way. Even setting GeometryDtype.na_value to None instead of NaN doesn't actually let this test pass

Reply:
I guess this seems to stem from `None` not being a "valid" fill_value in `obj_ser.unstack(level=level, fill_value=data.dtype.na_value)` (it is the default value, and deep down the call stack it is converted to nan in this case). The duplication is not ideal but as you say I can't see a good way around it. 

I think the answer is no, but would there be any value in xfailing the original test and adding this as `test_unstack_copy_for_gpd_compat`. This would at least tell us if the upstream test improved - but it seems unlikely for that to happen unless raised by us.